wgpu/backends/wgpu_native/_api.py

```diff
@@ -547,12 +547,13 @@ def get_current_texture(self):
         # * return the existing texture
         # * warn about it, and create a new one
         # * raise an error
-        # Right now we do the warning, so things still (kinda) keep working
+        # Right now we return the existing texture, so user can retrieve it in different render passes withing the same frame.
         if self._texture:
-            self._drop_texture()
-            logger.warning(
-                "get_current_texture() is called multiple times before pesent()."
-            )
+            return self._texture
+            # self._drop_texture()
+            # logger.warning(
+            #     "get_current_texture() is called multiple times before pesent()."
+            # )
```

Comment:
@almarklein can you comment on this change? is it OK?

Reply:
My motivation for the earlier behavior is to force a user to use pairs of `get_current_texture()` and `present()`, and that errors in user-code would be detected this way.

But I think @panxinmiao has a good case that just returning the existing texture provides more flexibility (for cases where another subsystem must render something), which would otherwise be cumbersome due to passing the current texture around ...

But let's remove the old commented code.